plugins/inputs/cloud_pubsub/cloud_pubsub.go

```diff
@@ -286,6 +305,41 @@ func (ps *PubSub) getGCPSubscription(subID string) (subscription, error) {
 	return &gcpSubscription{s}, nil
 }
 
+func (ps *PubSub) Init() error {
+	if ps.Subscription == "" {
+		return fmt.Errorf(`"subscription" is required`)
+	}
+
+	if ps.Project == "" {
+		return fmt.Errorf(`"project" is required`)
+	}
+
+	if ps.ContentEncoding == "" {
+		ps.ContentEncoding = "identity"
+	}
+
+	switch ps.ContentEncoding {
+	case "identity", "gzip":
+		// do nothing, encoding is valid
+	default:
+		return fmt.Errorf("invalid value for content_encoding")
+	}
+
+	if ps.ContentEncoding != "identity" {
+		var err error
+		ps.decoder, err = internal.NewContentDecoder(ps.ContentEncoding)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I was thinking of
```suggestion
	switch ps.ContentEncoding {
	case "", "identity":
		ps.ContentEncoding = "identity"
	case "gzip":
		var err error
		ps.decoder, err = internal.NewContentDecoder(ps.ContentEncoding)
		if err != nil {
			return err
		}
	default:
		return fmt.Errorf("invalid value %q for content_encoding", ps.ContentEncoding)
	}
```
which is more compact and easier to read IMO. :-)

Reply:
I agree this is better